### DISCUSSION TITLE
Dokku Linode Deploy SSH issue

### DISCUSSION THREAD

**Discussion by KayJay89** [dokku/dokku#4571] [Q&A] [answered]
**edit: This can be closed, I have started from scratch on a clean VPS and now it works. I suppose I have done something wrong prior to starting the tutorial. My apologies!**

Hi gals and boys,

Just a heads up: this is probably an "issue" linked to SSH and not so much Dokku but as I am encountering this when following the Dokku getting started example I thought it was best to ask here.

I discovered this wonderful PaaS only recently and have been trying to follow the Getting Started tutorial on my Linode VPS. Everything works fine until I try to deploy this particular getting started app.

When I try to run: ```git push dokku main:master``` I get requested for my password. I have tried both my root and my user password but none seem to work. I am a bit lost as to what I should do in this case.

When I read further I saw the note:
```
Note: Your private key should be registered with ssh-agent in your local development environment.
If you get a permission denied error when pushing, you can register your private key as 
follows: ssh-add -k ~/<your private key>.
```

But I always thought I should not give my private key but use my public one? My experience in using SSH keys is limited to adding them on my gitlab account and my Linode. I have no other experience with this so I could be completely confused because of my lack of knowledge.

When I run ```dokku ssh-keys:list``` my SSH key is present.

**Comment by josegonzalez** [comment, marked as answer]
That `ssh-add` command is meant to be run on your local machine, not on the remote. Sometimes your key isn't "registered" and so isnt used by a git push.
